### PR TITLE
Syndicate communication/services for sale

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Central Command have reported that various communication encryption
keys have been stolen. The use of these keys could HYPOTHETICALLY allow
faking of secure Central Command messages, or disruption of intelligence
dispatches. Central Command reassure all Nanotrasen stations that the
chances of this happening are very slim.
add: Adds bootleg announcer to uplink, costs 5 TC, fake a Central
Command message!
add: Adds intel disruption device to uplink, costs 10 TC, fake an "all
clear message" to the station, and watch as the extended suicides pile up.
/:cl:

ROB ROBINSON IS JUST TOO STEALTHY, WE CAN'T FIND HIM.

This is a good idea because traitors TC trade for Centcom dispatches
already, why not formalise it?

### CODE REVIEW

**Comment by ChangelingRain** [comment]
Potential major administration issue right the fuck here.

**Comment by calzilla1** [comment]
Sounds like fun

**Comment by Robustin** [comment]
Ive had this thought before but it really fucks with the ability of admins to guide a round with a Centcom message. All you need is one message at Roundstart going "We're taking you off the grid, no more space law, no more rules, kill anyone for any reason!" and you've got a guaranteed recipe for shit.

**Comment by GunHog** [comment]
Central Command messages are official communications from admins to players. Altering or disrupting this in any way robs the admins of communicating to the players in this manner. It is no longer official at that point.

Fake preventing automatic code blue to hide antags effectively circumvents its purpose. Its meaning will now be changed from "The game mode is Extended" to "The game mode is either Extended, Traitor, Traitor + Changeling, or Double Agents". Due to those game modes constituting half of our round probabilities, your change almost negates the original intent of the message: An IC way of announcing Extended.

**Comment by kevinz000** [comment]
Centcom messages should be an IC version of admin global announce.
Please don't make it unreliable.

**Comment by Buggy123** [comment]
On one hand, the people above me are right. On the other hand, I like the idea of adding more chaos to the game.

👍

**Comment by coiax** [comment]
What if I changed it so all messages have to be approved (or explicitly rejected) by online admins to be sent, and make it refundable? I can remove the "fake extended" if that's an issue.

**Comment by coiax** [comment]
Also, re "IC announcement of extended", a lot of extended shifts the admins do memes anyway, which ruins the sanctity of the extended announcement anyway.

**Comment by Buggy123** [comment]
That could work, maybe increase the cost a LOT too? With all the fuckery you can cause with this, at least 10tc might not be unreasonable.

**Comment by coiax** [comment]
Just checked, Baystation traitors have this, and it's hard coded to cost half of their TC.

**Comment by ExcessiveUseOfCobblestone** [comment]
Did you really just say the B word

**Comment by Chnkr** [comment]
You might want to make sure you use `stripped_multiline_input` to strip HTML from the report ;-)

**Comment by PKPenguin321** [comment]
Ruins the point of the extended announcement.

Maybe make them send a message to admins to approve/deny, like captain's station charter. Also make them refundable.

**Comment by Iamgoofball** [comment]
Ditch the fakextended but keep the announcer thing with the admin confirmation + refund if denied setup

File: code/game/objects/items/announcer.dm [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+	var/datum/effect_system/spark_spread/sparks = new
+	sparks.set_up(2, 0, src)
+	sparks.attach(src)
+	sparks.start()
+
+	qdel(src)
+
+/proc/create_command_report(mob/user)
+	var/input = input(usr, "Please enter anything you want. Anything. Serious.", "What?", "") as message|null
```
Comment (optimumtact) [inline code/game/objects/items/announcer.dm]: stripped input please

Comment (Jordie0608) [inline code/game/objects/items/announcer.dm]: I'd prefer if this remained as is just for admins, we do sometimes use html elements in the reports for extra flavor and trust that nobody is going to be dumb enough to do something nasty with this.

Comment (optimumtact) [inline code/game/objects/items/announcer.dm]: then you need to provide two seperate paths becaus trusting antags to "not be dumb enough to do something nasty" is exactly how our code was so leaky in the first place

Comment (Jordie0608) [inline code/game/objects/items/announcer.dm]: I meant trusting admins, not antags.

**Comment by RandomMarine** [comment]
We'll get maybe one or two weeks of fake centcom messages causing chaos until people will just start automatically assuming any trouble-inviting message is from some traitor.

Then from that point on it'll just be bored tator tots just global announcing whatever stupid jokes come to mind.

Fake extended reports defeat the entire point of why it was added in the first place. And people will just stop trusting the message entirely.

**Comment by Haevacht** [comment]
Don't make the extended message unreliable. Or actual centcomm for that matter.

**Comment by RandomMarine** [comment]
Another point. Admin-made centcom messages can be an invitation to bend the rules under some circumstances. I can't think of any examples offhand where this has happened, but bear with me.

Would a traitor making a faked "GAS ALL THE LIZARDS NOW" message mean staff is now valid to follow up on that?
If yes. Then would it really be healthy for traitors to spend a quarter of their TCs to bend the rule of valids to their will?
If no. Then wouldn't it mean that admin centcom messages will no longer be any excuse to do fun but otherwise invalid things?

**Comment by Steelpoint** [comment]
I think someone mistook the name 'security' for 'admins' in the 'Antags, Security and Admin Policy' thread and this is a attempt to get the crew to not trust the admins ever again.

👎

**Comment by coiax** [comment]
Closing to rework.

**Pull request closed without merging**